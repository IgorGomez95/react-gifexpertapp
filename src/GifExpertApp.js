import { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import GifGrid from './components/GifGrid';

// rafc snippet 
const GifExpertApp = () => {
    const [categories, setCategories] = useState(['Dragon Ball']);

    // const handleAdd = () => {
    //     setCategories( [...categories, 'nueva categor√≠a'] );
    // };
    
    return (
        <>
            <h2>Gift Expert App</h2>
            <AddCategory setCategories={ setCategories }/>
            <hr/>

            <ol>
                {
                    categories.map( category => (
                        <GifGrid 
                            key={ category }
                            category={ category }
                        />
                    ))
                }
            </ol>
        </>
    );
};

export default GifExpertApp;

