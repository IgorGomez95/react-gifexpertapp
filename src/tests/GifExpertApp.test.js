import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe('pruebas para el componente GifExpertApp', () => {
    test('debe de mostrar el componente correctamente ', () => {
        const wrapper = shallow( <GifExpertApp /> );
        expect(wrapper).toMatchSnapshot();
    });
    
    test('debe de mostrar una lista de categorÃ­as ', () => {
        const categories = ['Chelsea','Dragon Ball'];
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories } /> );
        
        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe(categories.length);
    })
    
});
