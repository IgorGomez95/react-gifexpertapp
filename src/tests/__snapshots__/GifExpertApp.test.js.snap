// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas para el componente GifExpertApp debe de mostrar el componente correctamente  1`] = `
<Fragment>
  <h2>
    Gift Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`pruebas para el componente GifExpertApp debe de mostrar una lista de categor√≠as  1`] = `
<Fragment>
  <h2>
    Gift Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Chelsea"
      key="Chelsea"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
